crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -87,12 +94,12 @@ impl SimpleFaucet {
         })
     }
 
-    async fn prepare_gas_coin(&self, total_amount: u64, uuid: Uuid) -> anyhow::Result<ObjectID> {
+    async fn prepare_gas_coin(&self, total_amount: u64, uuid: Uuid) -> GasCoinResponse {
```

Comment:
technically we can leave this as anyhow::Result as long as the error was return before `consumer.recv()`, but removing it make it easier to enforce returning a GasCoinResponse, and no need to handle the errs on the caller side.